app/src/main/java/org/oppia/app/topic/TopicActivity.kt

```diff
@@ -9,19 +9,25 @@ import org.oppia.app.player.exploration.ExplorationActivity
 import org.oppia.app.story.StoryActivity
 import org.oppia.app.topic.conceptcard.ConceptCardFragment
 import org.oppia.app.topic.questionplayer.QuestionPlayerActivity
+import org.oppia.domain.topic.TEST_TOPIC_ID_0
 import javax.inject.Inject
 
-/** The activity for tabs in Topic. */
+const val TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY = "TopicActivity.topic_id"
+
+/** The activity for displaying [TopicFragment]. */
 class TopicActivity : InjectableAppCompatActivity(), RouteToQuestionPlayerListener, RouteToConceptCardListener,
   RouteToTopicPlayListener, RouteToStoryListener, RouteToExplorationListener {
-
+  private lateinit var topicId: String
   @Inject
   lateinit var topicActivityPresenter: TopicActivityPresenter
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     activityComponent.inject(this)
-    topicActivityPresenter.handleOnCreate()
+    topicId  = checkNotNull(intent?.getStringExtra(org.oppia.app.topic.TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY)?: TEST_TOPIC_ID_0) {
+      "Expected topic ID to be included in arguments for TopicFragment."
```

Comment:
As this code stands, I don't think it will ever throw. I think you need to remove the elvis operator if you want it to fail when there's no result (at the moment it will default to TEST_TOPIC_ID_0.

Reply:
Resolved